### ISSUE TITLE
false positive "A form label must be associated with a control"

### ISSUE THREAD

**Issue by bona-petite** [jsx-eslint/eslint-plugin-jsx-a11y#759]
I'm using airbnb eslint configuration, and eslint complains when input's label is wrapped in another element:
```
function App() {
  return (
    <form>
      <h2>
        <label htmlFor="new-todo-input">
          What needs to be done?
        </label>
      </h2>
      <input
        type="text"
        id="new-todo-input"
        name="text"
      />
    </form>
  );
}
```
I'm getting this error during compilation:
```
Failed to compile.
src\App.jsx
  Line 5:9:  A form label must be associated with a control  jsx-a11y/label-has-associated-control
```

**Comment by manavm1990** [comment]
<img width="666" alt="image" src="https://user-images.githubusercontent.com/3387791/100285949-75250800-2f37-11eb-8b5d-8e19a36261b1.png">
 Me 2.

**Comment by ljharb** [comment]
This isn't a false positive. The airbnb config requires that every input be BOTH for-id linked, and also the input nested inside the label.

**Comment by manavm1990** [comment]
Thanks @ljharb I C [this has come up in the past.](https://github.com/airbnb/javascript/issues/1885#issuecomment-413349185)

**Comment by kalm42** [comment]
The error message needs to be updated. The form label *IS* associated with a control. Please be more specific that "A label must be the parent of the associated control".

**Comment by ljharb** [comment]
It’s not associated in every engine, though, unless it’s both nested and linked. A PR to improve the message would be welcome.

**Comment by liby** [comment]
> This isn't a false positive. The airbnb config requires that every input be BOTH for-id linked, and also the input nested inside the label.

@ljharb A false positive occurred when working with `Field` of Formik:
<img width="867" alt="image" src="https://user-images.githubusercontent.com/38807139/172313437-93e600be-d139-4b60-81bf-c097ed0210ec.png">

**Comment by ljharb** [comment]
@liby did you configure the plugin so it knows that a Field is an input?

**Comment by ljharb** [comment]
Then the warning is expected, because it’s not an input, it’s a Field. I’m not sure why it works elsewhere without that config.

**Comment by ljharb** [comment]
@liby at this point i think a new issue would be warranted; but first make sure you're using the latest version of the plugin, and you're also looking at the docs not on main, but on the version tag you're using.
